flow/project.py

```diff
@@ -291,13 +304,19 @@ def __init__(self, id, name, job, cmd, directives=None):
         self.directives._keys_set_by_user = keys_set_by_user
 
     def __str__(self):
-        return "{}({})".format(self.name, self.job)
+        if len(self._jobs) == 1:
+            return "{}({})".format(self.name, str(self._jobs[0]))
+        else:
+            minimum = self._jobs[0]._project.min_len_unique_id()
+            return "{}-{}: ({}-{})".format(
+                self.name, len(self._jobs), str(self._jobs[0])[0:minimum],
+                str(self._jobs[-1])[0:minimum])
```

Comment:
Yes, I agree that this isn't unique.
Should we also provide `_JobOperation.id` so that users could look up in the `.aggregates/` directory to get details of specific aggregate for an operation?

Reply:
Yes, especially if that makes it unique.

Reply:
@bdice I gave this a thought and I don't think that we'll need this to be unique.
Since `_JobOperation` is private hence users will never be able to print it by themselves.
Moreover, this is mainly used in scripts and we already print the script in this manner

```
Submitting cluster job 'playing-with/ee5ca9ab-6704a2f9/50/compute_sum/0000/c7033c1762ed0658882bc75c23dda4d5':
 - Group: compute_sum-50: (ee5-670)
```
 Which makes a submission of `_JobOperation` readable as well as unique.